net-p2p/bitcoin-core/bitcoin-core-25.0.ebuild

```diff
@@ -0,0 +1,340 @@
+		>=dev-qt/qtnetwork-5.15.5:5
+		>=dev-qt/qtwidgets-5.15.5:5
+		dbus? ( >=dev-qt/qtdbus-5.15.5:5 )
+	)
+	sqlite? ( >=dev-db/sqlite-3.38.5:= )
+	system-leveldb? ( virtual/bitcoin-leveldb )
+	system-libsecp256k1? ( >=dev-libs/libsecp256k1-0.3.1:=[recovery,schnorr] )
+	upnp? ( >=net-libs/miniupnpc-2.2.2:= )
+	zeromq? ( >=net-libs/zeromq-4.3.4:= )
+"
+DEPEND="
+	${RDEPEND}
+	systemtap? ( >=dev-util/systemtap-4.8 )
+"
+BDEPEND="
+	virtual/pkgconfig
+	daemon? (
+		acct-group/bitcoin
+		acct-user/bitcoin
+	)
+	qt5? ( >=dev-qt/linguist-tools-5.15.5:5 )
+	test? ( ${PYTHON_DEPS} )
+"
+IDEPEND="
+	qt5? ( dev-util/desktop-file-utils )
+"
+
+DOCS=(
+	doc/bips.md
+	doc/bitcoin-conf.md
+	doc/descriptors.md
+	doc/files.md
+	doc/i2p.md
+	doc/JSON-RPC-interface.md
+	doc/multisig-tutorial.md
+	doc/p2p-bad-ports.md
+	doc/psbt.md
+	doc/reduce-memory.md
+	doc/reduce-traffic.md
+	doc/release-notes.md
+	doc/REST-interface.md
+	doc/tor.md
+)
+
+PATCHES=(
+	"${FILESDIR}/25.0-syslibs.patch"
+	"${FILESDIR}/init.patch"
+)
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	! use system-leveldb || rm -r src/leveldb || die
+	if use system-libsecp256k1 ; then
+		rm -r src/secp256k1 || die
+		sed -e '/^DIST_SUBDIRS *=/s/\bsecp256k1\b//' -i src/Makefile.am || die
+	else
+		pushd src/secp256k1 >/dev/null || die
+		AT_NOELIBTOOLIZE=yes eautoreconf
+		popd >/dev/null || die
+	fi
+	eautoreconf
+
+	# we say --disable-util-util, so we can't test bitcoin-util
+	sed -ne '/^  {/{h;:0;n;H;/^  }/!b0;g;\|"exec": *"\./bitcoin-util"|d};p' \
+		-i test/util/data/bitcoin-util-test.json || die
+
+	sed -e 's/^\(complete -F _bitcoind\b\).*$/\1'"$(usev daemon ' bitcoind')$(usev qt5 ' bitcoin-qt')/" \
+		-i contrib/completions/bash/bitcoind.bash-completion || die
+}
+
+src_configure() {
+	local wallet ; if use berkdb || use sqlite ; then wallet=enable ; else wallet=disable ; fi
+	local myeconfargs=(
+		--disable-static
+		--${wallet}-wallet
+		$(use_with sqlite)
+		$(use_with berkdb bdb)
+		$(use_enable systemtap usdt)
+		$(use_with upnp miniupnpc)
+		$(use_with nat-pmp natpmp)
+		$(use_enable test tests)
+		--disable-bench
+		--disable-fuzz{,-binary}
+		$(use_with qrcode qrencode)
+		--disable-ccache
+		$(use_enable asm)
+		$(use_enable zeromq zmq)
+		$(use_enable man)
+		$(use_enable external-signer)
+		--with-utils
+		$(use_enable {bitcoin,util}-cli)
+		--enable-util-tx
+		--${wallet}-util-wallet
+		--disable-util-util
+		$(use_with libs)
+		$(use_with daemon)
+		$(use_with qt5 gui qt5)
+		$(use_with dbus qtdbus)
+		$(use_with system-leveldb)
+		$(use_with system-libsecp256k1)
+	)
+	econf "${myeconfargs[@]}"
+}
+
+src_install() {
+	use external-signer && DOCS+=( doc/external-signer.md )
+	use berkdb || use sqlite && DOCS+=( doc/managing-wallets.md )
+	use libs && DOCS+=( doc/shared-libraries.md )
+	use systemtap && DOCS+=( doc/tracing.md )
+	use zeromq && DOCS+=( doc/zmq.md )
+
+	default
+
+	! use libs || find "${ED}" -name '*.la' -delete || die
```

Comment:
And particular reason the .la files are not unconditionally deleted? Also please consider adding `-type f` to find.

Reply:
> An[y] particular reason the .la files are not unconditionally deleted?

No `.la` files are installed unless `use libs`. There is no reason to make a pass over the directory tree to look for files that don't exist. That said, the inode and dentry caches are likely already to be warm at this point, so a needless check likely wouldn't do much harm.

> Also please consider adding `-type f`

Will do. Note, I didn't originate this line; I merely incorporated it from…

https://github.com/gentoo/gentoo/blob/373c7c27b7564a09496dab34d369ea8424167b59/net-libs/libbitcoinconsensus/libbitcoinconsensus-24.0.1.ebuild#L93